app/models/grda_warehouse/tasks/sanity_check_service_history.rb

```diff
@@ -70,6 +69,17 @@ def sanity_check
       end
     end
 
+    def client_still_processing?(client_id)
+      job_ids = Delayed::Job.where(queue: ::ServiceHistory::RebuildEnrollmentsByBatchJob.queue_name, failed_at: nil).
+        jobs_for_class('ServiceHistory::RebuildEnrollments').
+        pluck(:id)
+      enrollment_ids = GrdaWarehouse::Hud::Client.where(id: client_id).
+        joins(source_enrollments: :project).
+        pluck(Arel.sql(e_t[:id].as('enrollment_id').to_sql))
+
+      GrdaWarehouse::Hud::Enrollment.where(id: enrollment_ids, service_history_processing_job_id: job_ids).exists?
+    end
```

Comment:
I think this whole method can be made cleaner.  The query on line 80 should be rolled into the one on 76, just where(e_t[:service_history_processing_job_id].in(job_ids))`